### ISSUE TITLE
Bash completion directory is missing after updating the stable toolchain to 1.71.0

### ISSUE THREAD

**Issue by tristan957** [rust-lang/rustup#3407]
### Problem

When launching a terminal, I am now seeing the following error:

```
/home/tristan957/.local/share/bash-completion/completions/cargo: line 2: /home/tristan957/.local/share/rustup/toolchains/stable-x86_64-unknown-linux-gnu/etc/bash_completion.d/cargo: No such file or directory
```

```
 $ echo $RUSTUP_HOME
/home/tristan957/.local/share/rustup
```

I am not entirely sure how this might've happened or if I am the only one experiencing this issue. The 1.70 toolchain that I have installed still has the bash completion files.

### Steps

I just updated from 1.70 to 1.71, and this started happening.

### Rustup version

```console
rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.71.0 (8ede3aae2 2023-07-12)`
```

### Installed toolchains

```console
$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/tristan957/.local/share/rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
1.70.0-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.71.0 (8ede3aae2 2023-07-12)
```

**Comment by ehuss** [comment]
Thanks for the report! This seems to have been a regression that was fixed a few days ago, but wasn't backported. I have followed up on https://github.com/rust-lang/rust/pull/113579 to propose getting it backported.

**Comment by tristan957** [comment]
@ehuss thanks for getting back to me. I hope to see a backport!

**Comment by iwanders** [comment]
I've been using this as a workaround:
```
source  ~/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/src/etc/bash_completion.d/cargo
```

**Comment by mcint** [comment]
This is fixed as of `1.71.1`. Thanks! This was bugging me since I set up my new laptop. Nixpkg installation of rustup that I hadn't written a local patch for.

**Comment by tristan957** [comment]
Can confirm.
